libPhoneNumber/NBPhoneNumberOfflineGeocoder.h

```diff
@@ -0,0 +1,40 @@
+//
+//  NBPhoneNumberOfflineGeocoder.h
+//  libPhoneNumberiOS
+//
+//  Created by Rastaar Haghi on 6/12/20.
+//  Copyright © 2020 ohtalk.me. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "NBGeocoderMetadataHelper.h"
+#import "NBPhoneNumber.h"
+#import "NBPhoneNumberDefines.h"
+#import "NBPhoneNumberUtil.h"
+NS_ASSUME_NONNULL_BEGIN
+
+@interface NBPhoneNumberOfflineGeocoder : NSObject
+
+- (instancetype)init;
+
+- (NSString *)descriptionForValidNumber:(NBPhoneNumber *)phoneNumber
+                       withLanguageCode:(NSString *)language;
+- (NSString *)descriptionForValidNumber:(NBPhoneNumber *)phoneNumber
+                       withLanguageCode:(NSString *)language
+                         withUserRegion:(NSString *)userRegion;
+- (NSString *)descriptionForNumber:(NBPhoneNumber *)phoneNumber
+                  withLanguageCode:(NSString *)languageCode;
+- (NSString *)descriptionForNumber:(NBPhoneNumber *)phoneNumber
+                  withLanguageCode:(NSString *)languageCode
+                    withUserRegion:(NSString *)userRegion;
+@end
+
+@interface NBPhoneNumberOfflineGeocoder ()
+
+- (NSString *)descriptionForNumber:(NBPhoneNumber *)phoneNumber;
+- (NSString *)descriptionForNumber:(NBPhoneNumber *)phoneNumber
+                    withUserRegion:(NSString *)userRegion;
+
+@end
```

Comment:
Maybe you don't need to make it an extension. Just make it part of the main API, but maybe add something like:

`#pragma mark - Convenience methods`

to separate them from the main ones

Reply:
Done.